src/declarativebookmarkmodel.cpp

```diff
@@ -47,12 +51,12 @@ void DeclarativeBookmarkModel::removeBookmark(const QString& url)
         return;
     }
 
-    if (bookmarks.contains(url)) {
-        int index = bookmarkUrls.indexOf(url);
+    if (bookmarkIndexes.keys().contains(url)) {
```

Comment:
And above bookmarkIndex function could be used in DeclarativeBookmarkModel::contains(const QString& url). However, we'd get a performance hit when comparing to map.